common-react/@dbeaver/ui-kit/src/Checkbox/Checkbox.tsx

```diff
@@ -6,15 +6,64 @@
  * you may not use this file except in compliance with the License.
  */
 
-import { Checkbox as AriaCheckbox, type CheckboxProps as AriaKitCheckboxProps } from '@ariakit/react';
+import { Checkbox as AriaCheckbox, VisuallyHidden, type CheckboxProps as AriaKitCheckboxProps } from '@ariakit/react';
+import { useState } from 'react';
+import './Checkbox.css';
+import type { ControlSize } from '../types/controls.js';
 
-export interface CheckboxProps extends AriaKitCheckboxProps {}
+export interface CheckboxProps extends Omit<AriaKitCheckboxProps, 'render' | 'size'> {
+  size?: ControlSize;
+  icon?: React.ReactNode;
+  indeterminate?: boolean;
+  indeterminateIcon?: React.ReactNode;
+}
 
-export function Checkbox({ children, ...props }: CheckboxProps) {
+export function Checkbox({ children, className, icon, indeterminate, indeterminateIcon, size = 'medium', ...props }: CheckboxProps) {
+  const [checked, setChecked] = useState(props.defaultChecked ?? props.checked ?? false);
+  const [focusVisible, setFocusVisible] = useState(false);
```

Comment:
This is not correctly synchronized code. Control may be controlled and not controlled; therefore, "controlling" props must be prioritized over the inner state.

```
const [innerChecked, setInnerChecked] = ...
const checked = props.checked ?? innerChecked;
```

Reply:
Thank you!